docs/features/video-protection.md

```diff
@@ -0,0 +1,108 @@
+# Video Protection
+
+Videos can be protected in JW platform in two ways:
+
+- **Signed URLs:** A player can only **access** video URLs from JW Player backend using a time-bound JWT token
+- **DRM:** A player can only **play** videos using a time-bound decryption key. 
+
+The  JW Web Player, as well as the other SDKs, supports these mechanismns out-the-box. However this require a server side entitlement service which is **NOT** part of the web app and needs to be **custom developed**.
+
+This article outlines such entitlement service should work. 
+
+## Signed URLs
+
+With [URL signing](https://support.jwplayer.com/articles/how-to-enable-url-token-signing) enabled on the JW property, a video client can only access the media URLs from JW backends when it has a valid JWT token:
+
+```
+GET media/PEEzDfdA?token=<tokenA>
+{
+ "title":"Video Title",
+ "description":"Lorem ipsum dolor sit amet", 
+ "sources":[https://content.jwplatform.com/manifests/PEEzDfdA.m3u8?token=<tokenB>]
+}
+```
+
+Missing or invalid tokens will result in a `403` forbidden.
+
+Notice that the API response will also have all video URLs signed, making subsequent calls accessing the video sources easy.
+
+### Token generation
+
+This is what a decoded token looks like:
+
+```
+{
+  "resource": "/v2/media/Xw0oaD4q",
+  "exp": 1893456000
+}
+```
+
+Notice that every token grants access to a unique resource. It's not possible to have token that grants access to a collection of resources.
+
+The tokens are generated using the property API key. See [the documenation](https://developer.jwplayer.com/jwplayer/docs/protect-your-content-with-signed-urls) on how to generate a token.
+
+## DRM
+
+JW [supports](https://developer.jwplayer.com/jwplayer/docs/enable-drm-with-jw-stream) three DRM systems
+
+* Widevine (Google ecosystem)
+* PlayReady (Microsoft ecosystem)
+* Fairplay (Apple ecosystem)
+
+These systems require a time-bound decryption key, which can be fetched from the license URL. 
+
+Also all DRM URLs require to be signed with a token.
+
+```
+GET media/PEEzDfdA/drm/:drm_policy_id?token=<tokenA>
+{
+ "title":"Video Title",
+ "description":"Lorem ipsum dolor sit amet", 
+ "sources":[{
+   "drm":{
+      "file":":https://content.jwplatform.com/manifests/PEEzDfdA.m3u8?token=<tokenB>",
+      "widevine":{"url":"<widevine_license_url>?token=<tokenC>"}
+}]
+}
+```
+
+The full details of this endpoint can be found [here](https://developer.jwplayer.com/jwplayer/reference/get_v2-media-media-id-drm-policy-id).
+
+## Custom Entitlement Service
```

Comment:
I would avoid trying to give specific details for this, because like we talked about in our brainstorm it can get really complicated really fast.  You can present the 3 models as 'best practices' with a disclaimer that every customer situation is likely to be different.

Reply:
Ok